immutable.go

```diff
@@ -757,11 +757,10 @@ func (m *Map[K, V]) Set(key K, value V) *Map[K, V] {
 // This function will return a new map even if the updated value is the same as
 // the existing value because Map does not track value equality.
 func (m *Map[K, V]) SetMany(entries map[K]V) *Map[K, V] {
-	n := m.clone()
 	for k, v := range entries {
-		n.set(k, v, true)
+		m = m.Set(k, v)
```

Comment:
Same as for sets, I'd avoid calling Set (because a new Map is created for each entry).

I think this one ought to be longer, similar to `set()` (with modify=true), but with a for loop. I can offer something in a day or so if it's not clear.

Reply:
Looking forward to it. :slightly_smiling_face:

Reply:
In the end I did something simpler than what I expected - i.e. to use `mutable = false` for first item, and then `mutable=true` for the rest. See https://github.com/benbjohnson/immutable/pull/40